okta/resource_user_basic_schema.go

```diff
@@ -0,0 +1,177 @@
+	return &schema.Resource{
+		Create: resourceUserBaseSchemaCreate,
+		Read:   resourceUserBaseSchemaRead,
+		Update: resourceUserBaseSchemaUpdate,
+		Delete: resourceUserBaseSchemaDelete,
+		Exists: resourceUserBaseSchemaExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"index": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Subschema unique string identifier",
+				ForceNew:    true,
+			},
+			"title": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Subschema title (display name)",
+			},
+			"type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"string", "boolean", "number", "integer", "array", "object"}, false),
+				Description:  "Subschema type: string, boolean, number, integer, array, or object",
+				ForceNew:     true,
+			},
+			"array_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"string", "number", "interger", "reference"}, false),
+				Description:  "Subschema array type: string, number, interger, reference. Type field must be an array.",
+				ForceNew:     true,
+			},
+			"permissions": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"HIDE", "READ_ONLY", "READ_WRITE"}, false),
+				Description:  "SubSchema permissions: HIDE, READ_ONLY, or READ_WRITE.",
+				Default:      "READ_ONLY",
+			},
+			"master": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				// Accepting an empty value to allow for zero value (when provisioning is off)
+				ValidateFunc: validation.StringInSlice([]string{"PROFILE_MASTER", "OKTA", ""}, false),
+				Description:  "SubSchema profile manager, if not set it will inherit its setting.",
+			},
+		},
+	}
+}
+
+func resourceUserBaseSchemaCreate(d *schema.ResourceData, m interface{}) error {
+	if err := updateBaseSubschema(d, m); err != nil {
+		return err
+	}
+	d.SetId(d.Get("index").(string))
+
+	return resourceUserBaseSchemaRead(d, m)
+}
+
+func resourceUserBaseSchemaExists(d *schema.ResourceData, m interface{}) (bool, error) {
+	subschema, err := getBaseSubSchema(d, m)
+
+	return subschema != nil, err
+}
+
+func resourceUserBaseSchemaRead(d *schema.ResourceData, m interface{}) error {
+	subschema, err := getBaseSubSchema(d, m)
+	if err != nil {
+		return err
+	} else if subschema == nil {
+		// See https://github.com/articulate/terraform-provider-okta/issues/144
+		// Occassionally a schema prop would be created and the read would not find it.
+		// There appears to be a delay of availability on the Okta side, thus the backoff.
+		fmt.Println("Could not find an existence subschema property, backing off and retrying. Known timing issue")
```

Comment:
We can remove this retry logic for base schemas.

Reply:
yes) of course